src/funcs.c

```diff
@@ -441,12 +443,12 @@ Obj             EvalFunccall0args (
     /* evaluate the function                                               */
     func = EVAL_EXPR( FUNC_CALL( call ) );
 
+    /* call the function and return the result                             */
+    SET_BRK_CALL_TO( call );
     if (TNUM_OBJ(func) != T_FUNCTION) {
       return DispatchFuncCall(func, 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0 );
```

Comment:
We probably should (in this or in a future PR) change this return to `result = Dispatch...` and add a test case for the bug fixed by it (i.e. a "custom" func call which fails to return something.

Reply:
What I think you mean is that
```
if (TNUM_OBJ(func) != T_FUNCTION) {
        return DispatchFuncCall(func, 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0 );
}
result = CALL_0ARGS( func );
if (STATE(UserHasQuit) ... etc.
```
should be changed to 
```
if (TNUM_OBJ(func) != T_FUNCTION) {
     result = DispatchFuncCall(func, 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0, (Obj) 0 );
} else {
     result = CALL_0ARGS( func );
}
if (STATE(UserHasQuit) ... etc.
```
so that all of the same post-processing (checking of state, checking of return value) occurs in the user-level-defined call syntax case as in the older "standard" function calls. If so, that makes sense to me and I am sorry I didn't see that the first time, I was focused on the misleading backtrace messages. I'd be happy to make this change (and the seven parallel changes in the following) if you all would like, although a hint as to how to get my git state in sync with this pull request so I can add to it would be helpful, esp. since I messed up #1814 when I was supposed to rebase it...

Reply:
@gwhitney yes that's what I meant (typed this on a phone :-).

There is nothing for you to be sorry about or apologetic for. You noticed and issue and sent a PR to fix it: THANKS!!! And this then made another problem visible: thanks again! Please don't apologize for not doing even more work :-).

That said, of course further contributions from you are highly welcome :-).

To fix you git setup, there are various ways. I'll try to describe one. Roughly speaking, we'll first get the state represented in this pull request, and then fix you `fix/master/issue_917` branch, after which you could actually reopen your PR #1814 (and possibly work on more features.

0. First, I am assuming you are doing this in a terminal, either on a Unix-like operating system, or in cygwin; if you are using something else (e.g. bare Windows), I can't help much. Next, I assume for the following that you start out in the root of your clone of the `gap` repository.

1. Change to the master branch, and update:
```
git checkout master
git pull --ff-only
```
2. Update your `fix/master/issue_917` branch locally, by entering this command, which pulls the state of this PR and (force) updates `fix/master/issue_917`:
```
git fetch --force https://github.com/gap-system/gap pull/1821/head:fix/master/issue_917
```
3. Change to your branch, and make sure it works
```
git checkout fix/master/issue_917
# now perhaps make followed by running tests
```
4. If this all worked, re-open PR #1814, and then afterwards, update your branch on GitHub to match your new local version (WARNING: the following command assumes you configured your branch `fix/master/issue_917` to track its github counterpart. If not, git may not know where to push the changes. But since you managed to push before, you probably know how to do this (just add in the `--force` option)
```
git push --force fix/master/issue_917
```

That should be it. You can now also add further commits, and `git push` your branch again.